### ISSUE TITLE
Confusing about the generator definition at Chapter 6.3

### ISSUE THREAD

**Issue by tairen99** [fchollet/deep-learning-with-python-notebooks#14]
Dear author,
        In addition to the model prediction from Chapter 6.3 from the previous question: 
[ Next prediction for 6.3, temperature after data set ends? Also, testing set?](https://github.com/fchollet/deep-learning-with-python-notebooks/issues/5), I do not understand how the generator function is defined. 
In the supervised learning, we may want to train the model through the training data (pairs of predictors and responses). Like the regression example in Chapter 3.6, the predictor is a vector and the response is a numerical value. But in Chapter 6.3, 
         ` indices = range(rows[j] - lookback, rows[j], step)
            samples[j] = data[indices]
            targets[j] = data[rows[j] + delay][1] `
The predictor for a sample time point is a matrix and the response is a numerical value. 
I do not why you defined the regression in this way? I understand in the image classification as the example in Chapter 5.2, you use the generator to automatically generate the training data: a 3D figure corresponds to a label, so you do not need to store all the images in the memory.
            Meanwhile, why the **batch_size = 128**? because you want to make the **val_steps** as an integer? In the real application, how we chose a good batch_size?
          Thank you very much!
